server/master/src/main/java/org/apache/accumulo/master/metrics/Metrics2ReplicationMetrics.java

```diff
@@ -124,4 +142,55 @@ protected int getNumConfiguredPeers() {
   protected int getMaxReplicationThreads() {
     return replicationUtil.getMaxReplicationThreads(master.getMasterMonitorInfo());
   }
+
+  protected void addReplicationQueueTimeMetrics() {
+    // Exit early if replication table is offline
+    if (TableState.ONLINE != Tables.getTableState(master.getInstance(), ReplicationTable.ID)) {
+      return;
+    }
+
+    // Exit early if we have no replication peers configured
+    if (replicationUtil.getPeers().isEmpty()) {
+      return;
+    }
+
+    Set<Path> paths = replicationUtil.getPendingReplicationPaths();
+
+    // We'll take a snap of the current time and use this as a diff between any deleted
+    // file's modification time and now. The reported latency will be off by at most a
+    // number of seconds equal to the metric polling period
+    long currentTime = System.currentTimeMillis();
+
+    // Iterate through all the pending paths and update the mod time if we don't know it yet
+    for (Path path : paths) {
+      if (!pathModTimes.containsKey(path)) {
+        try {
+          pathModTimes.put(path, master.getFileSystem().getFileStatus(path).getModificationTime());
+        } catch (IOException e) {
+          // Ignore all IOExceptions
+          // Either the system is unavailable or the file was deleted
+          // since the initial scan and this check
+        }
+      }
+    }
+
+    // Remove all currently pending files
+    Set<Path> deletedPaths = new HashSet<>(pathModTimes.keySet());
+    deletedPaths.removeAll(paths);
+
+    // Exit early if we have no replicated files to report on
+    if (deletedPaths.isEmpty()) {
+      return;
+    }
+
+    replicationQueueTimeStat.resetMinMax();
+
+    for (Path path : deletedPaths) {
+      // Remove this path and add the latency
+      long modTime = pathModTimes.remove(path);
```

Comment:
This will throw an error is `pathModTimes` ever does not have a mapping for `path` because you're unboxing the `Long` into a `long`.

Reply:
I was actually thinking how best to handle this.  It'd be an implementation error if the path to be removed wasn't in `pathModTimes`, since `deletedPaths` is a subset of `pathModTimes`.  Maybe check for existence and log an error if it doesn't exist?

Reply:
It's just the unboxing you really have to worry about. Something like the following would be fine:

```java
Long modTime = pathModTimes.remove(path);
if (modTime != null) {
  ...
}
```

Java will handle the unboxing for you once you guaranteed that `modTime` is definitely non-null.